tests/bsim/bluetooth/mesh/src/test_beacon.c

```diff
@@ -908,13 +970,440 @@ static void test_rx_secure_beacon_interval(void)
 	PASS();
 }
 
+#if CONFIG_BT_MESH_V1d1
+static bool private_beacon_check(const uint8_t *net_id, void *ctx)
+{
+	bool ret;
+	bool same_random = (bool *)ctx;
+
+	if (memcmp(beacon.adv_addr.a.val, &last_beacon_adv_addr, BT_ADDR_SIZE) == 0) {
+		return false;
+	}
+
+	memcpy(&last_beacon_adv_addr, beacon.adv_addr.a.val, BT_ADDR_SIZE);
+
+	if (same_random) {
+		ret = memcmp(beacon.random, last_random, 13) == 0;
+	} else {
+		ret = memcmp(beacon.random, last_random, 13) != 0;
+	}
+
+	memcpy(&last_random, beacon.random, 13);
+
+	return ret;
+}
+
+static void provision(const struct bt_mesh_test_cfg *dev_cfg)
+{
+	int err;
+
+	err = bt_mesh_provision(net_key, 0, 0, 0, dev_cfg->addr, dev_cfg->dev_key);
+	if (err) {
+		FAIL("Provisioning failed (err %d)", err);
+		return;
+	}
+}
+
+static void test_tx_priv_on_iv_update(void)
+{
+	uint8_t status;
+	struct bt_mesh_priv_beacon val;
+	int err;
+
+	bt_mesh_test_cfg_set(NULL, WAIT_TIME);
+	bt_mesh_device_setup(&prov, &prb_comp);
+	provision(&tx_cfg);
+
+	val.enabled = 1;
+	val.rand_interval = random_interval;
+
+	err = bt_mesh_cfg_cli_beacon_set(0, tx_cfg.addr, 0, &status);
+	if (err || status != 0) {
+		FAIL("Beacon set failed (err %d, status %u)", err, status);
+	}
+
+	err = bt_mesh_priv_beacon_cli_set(0, tx_cfg.addr, &val);
+	if (err) {
+		FAIL("Failed to enable Private Beacon (err=%d)", err);
+		return;
+	}
```

Comment:
Could the source from this line until begin of the function put in the separate function? Seems there is copy paste here and some functions below.

Reply:
abstracted into `tx_priv_setup`